### ISSUE TITLE
Java errors in RuntimePlugin.java

### ISSUE THREAD

**Issue by Ingo60** [usethesource/impulse#2]
Trying to compile this under Luna, I get 3 errors:

```
Description Resource    Path    Location    Type
Name clash: The method getAdapter(Object, Class<T>) of type RuntimePlugin.ResourceToSourceEntityAdapter has the same erasure as getAdapter(Object, Class) of type IAdapterFactory but does not override it  RuntimePlugin.java  /impulse/src/io/usethesource/impulse/runtime    line 44 Java Problem
The method getAdapter(Object, Class<T>) of type RuntimePlugin.ResourceToSourceEntityAdapter must override or implement a supertype method   RuntimePlugin.java  /impulse/src/io/usethesource/impulse/runtime    line 44 Java Problem
The type RuntimePlugin.ResourceToSourceEntityAdapter must implement the inherited abstract method IAdapterFactory.getAdapter(Object, Class) RuntimePlugin.java  /impulse/src/io/usethesource/impulse/runtime    line 39 Java Problem
```

**Comment by jurgenvinju** [comment]
Yes, Mars is not backward compatible with Luna unfortunately. This change was accidental, but when compiling for Mars you can't know that it won't work on Luna. Do you see an opportunity for Frege to also move on to Mars, or is this a show stopper?

**Comment by Ingo60** [comment]
No, the point is I want to go to Mars while supporting Kepler and Luna also.

But it looks like I am not able to do this.

I have changed the `getAdapter` method back manually for this time, and it works fine on Kepler and Luna.
But on Mars, I still get the `setDefaultEditor` error, as before! 
Is there no way to do this without using internal APIs? Reflection, maybe?
For otherwise, I need to go to MARS.1 for plugin development?

This would be ok for a while, but I really hope this is it, then. For, my most precious ressource is development time, and I simply cannot afford to squander days with such stuff.

**Comment by Ingo60** [comment]
I "fixed" the setDefaultEditor problem by making sure it would never reach the method call, but always goes to the else branch.
Seems to work on Mars, as well as Luna. (However, the files have the ordinary text file icon in the project explorer).
